apps/remix-ide/src/app/files/fileManager.ts

```diff
@@ -227,22 +227,21 @@ class FileManager extends Plugin {
   */
   async writeMultipleFiles(filePaths, fileData, folderPath) {
     try {
-      let alert = true
-      for (let i = 0; i < filePaths.length; i++) {
-        const installPath = folderPath + "/" + filePaths[i]
+      filePaths.forEach(async (filePath, index) => {
```

Comment:
forEach is not a good place to do anything async. It is generally not recommended to do anything async in a foreach because forEach loops doesn't wait for any async ops to finish and the result of which is race conditions and bugs.
Using a for...of or a Promise.all might be better here but your milage may vary.